StateOfChargeMonitoring.c

```diff
@@ -0,0 +1,28 @@
+#include "Config/ParameterConfig.h"
+#include "StateOfChargeMonitoring.h"
```

Comment:
This file looks very similar to `TemperatureMonitoring.h`. Any way to reduce the duplication?